app_server/app/models/Storage.js

```diff
@@ -214,20 +229,79 @@ Class.getListOfUsers = function() {
 };
 
 /**
+ * @param  {string} userId - userid of the user who created stream
+ * @param  {object} streamAttributes
+ * @param  {string} streamAttributes.streamKey
+ * @param  {string} streamAttributes.roomId
+ * @return {Promise<Sequelize.object>}
+ */
+Class.createStream = function(userId, streamAttributes) {
+  var userPromise = this.models.User.findById(userId);
+  var streamPromise = this.models.Stream.create(streamAttributes);
+  var that = this;
+
+  return Promise.join(userPromise, streamPromise,
+    function(user, stream) {
+      return user.addStream(stream).then(function() {
+        return that.getStreamById(stream.streamId);
+      });
+    });
+};
```

Comment:
Also remember to implement `catch` clause somewhere in the pipeline. Either here or in upper service layer.